components/style/properties/longhand/inherited_box.mako.rs

```diff
@@ -40,73 +40,13 @@ ${helpers.single_keyword("color-adjust",
                          animatable=False)}
 
-<%helpers:longhand name="image-rendering" animatable="False">
-    pub mod computed_value {
-        use cssparser::ToCss;
-        use std::fmt;
-
-        #[allow(non_camel_case_types)]
-        #[derive(Copy, Clone, Debug, PartialEq)]
-        #[cfg_attr(feature = "servo", derive(HeapSizeOf, Deserialize, Serialize))]
-        pub enum T {
-            auto,
-            crispedges,
-            % if product == "gecko":
-                optimizequality,
-                optimizespeed,
-            % else:
-                pixelated,  // firefox doesn't support it (https://bugzilla.mozilla.org/show_bug.cgi?id=856337)
-            % endif
-        }
-
-        impl ToCss for T {
-            fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
-                match *self {
-                    T::auto => dest.write_str("auto"),
-                    T::crispedges => dest.write_str("crisp-edges"),
-                    % if product == "gecko":
-                        T::optimizequality => dest.write_str("optimizeQuality"),
-                        T::optimizespeed => dest.write_str("optimizeSpeed"),
-                    % else:
-                        T::pixelated => dest.write_str("pixelated"),
-                    % endif
-                }
-            }
-        }
-    }
-
-    use values::NoViewportPercentage;
-    impl NoViewportPercentage for SpecifiedValue {}
-
-    pub type SpecifiedValue = computed_value::T;
-
-    #[inline]
-    pub fn get_initial_value() -> computed_value::T {
-        computed_value::T::auto
-    }
-
-    pub fn parse(_: &ParserContext, input: &mut Parser) -> Result<SpecifiedValue,()> {
-        // According to to CSS-IMAGES-3, `optimizespeed` and `optimizequality` are synonyms for
-        // `auto`.
-        match_ignore_ascii_case! {
-            try!(input.expect_ident()),
-            "auto" => Ok(computed_value::T::auto),
-            "crisp-edges" => Ok(computed_value::T::crispedges),
-            % if product == "gecko":
-                "optimizequality" => Ok(computed_value::T::optimizequality),
-                "optimizespeed" => Ok(computed_value::T::optimizespeed),
-            % else:
-                "optimizequality" => Ok(computed_value::T::auto),
-                "optimizespeed" => Ok(computed_value::T::auto),
-                "pixelated" => Ok(computed_value::T::pixelated),
-            % endif
-            _ => Err(())
-        }
-    }
-
-    use values::computed::ComputedValueAsSpecified;
-    impl ComputedValueAsSpecified for SpecifiedValue { }
-</%helpers:longhand>
+// According to to CSS-IMAGES-3, `optimizespeed` and `optimizequality` are synonyms for `auto`.
+// And, firefox doesn't support 'pixelated' (https://bugzilla.mozilla.org/show_bug.cgi?id=856337)
+${helpers.single_keyword("image-rendering",
+                         "auto crispedges",
+                         extra_gecko_values="optimizequality optimizespeed",
```

Comment:
Okay, yeah, the problem lies in `crisp-edges`. I'll better add an option for that in python.